README.md

```diff
@@ -16,15 +17,138 @@ TerseTS can be compiled and cross-compiled from source:
+<details>
+<summary><strong>C/C++ Usage Example</strong></summary>
+
+```c 
+#include "tersets.h"
+#include <stdio.h>
+
+int main() {
+   double uncompressed_values[] = {1.0, 2.0, 3.0, 4.0, 5.0};
+   struct UncompressedValues uncompressed_values = {data, 5};
+   
+   printf("Uncompressed data length: %lu\n", uncompressed_values.len);
+   
+   // Configuration for compression. 
+   // The supported compression methods are specified in tersets.zig.
+   // Method 2 is SwingFilter and 0.1 error bound.
+   struct Configuration config = {2, 0.0}; 
+
+   // Prepare for compressed data. 
+   // The compressed values point to dynamically allocated data that should be deallocated.
+   struct CompressedValues compressed_values;
+    
+   // Compress the data.
+   int32_t result = compress(uncompressed_values, &compressed_values, config);
+   if (result != 0) {
+      printf("Compression failed with error code %d\n", result);
+      return -1;
+   }
+
+   printf("Compression successful. Decompressed data length: %lu\n", compressed_values.len);
+    
+   // Prepare for decompressed data. 
+   // The decompressed values point to dynamically allocated data that should be deallocated.
+   struct UncompressedValues decompressed_values;
+   
+   // Decompress the data.
+   int32_t result = decompress(compressed_values, &decompressed_values);
+   if (result != 0) {
+      printf("Decompression failed with error code %d\n", result);
+      return -1;
+   }
+
+   printf("Decompression successful. Decompressed data length: %lu\n", decompressed_values.len);
+    
+   // Free the compressed and decompressed value.
+   free(decompressed_values.data);
+   free(compressed_values.data);
+   return 0;
+}
+```
+
+TerseTS provides `./bindings/c/tersets.h` as API for C/C++ which should be included in the source code, i.e., `#include "tersets.h"`. You need to [link](#linking) the TerseTS library to the project. 
+
+Ensure that the method field in `Configuration` is set to a valid compression/decompression method supported by `TerseTS`.
+
+Free dynamically allocated memory appropriately to avoid memory leaks.
+</details>
+
+<a id="python-usage-example"></a>
+<details>
+<summary><strong>Python Usage Example</strong></summary>
+
+```python
+import random
+import sys
+from tersets import compress, decompress, Method
+
+uncompressed_values = [1.0, 2.0, 3.0, 4.0, 5.0]
+error_bound = 0.1
+
+print("Uncompressed data length: ", len(uncompressed_values))
+
+# Compress using the SwingFilter method.
+compressed_values = compress(uncompressed, 0.1, Method.SwingFilter)
+
+print("Compression successful. Compressed data length: ", len(compressed_values))
+
+# Decompress the data.
+decompressed_values = decompress(compressed_values)
+
+print("Decompression successful. Decompressed data length: ", len(decompressed_values))
+```
+
+TerseTS provides `./bindings/python/tersets/__init__.py` as binding for Python which allows directly importing tersets, i.e., `import tersets`.
```

Comment:
Maybe add an additional sentence stating that the [Python bindings automatically loads the native library](https://github.com/cmcuza/TerseTS/blob/doc/improve-readme/bindings/python/tersets/__init__.py#L50) but [assumes it is not moved](https://github.com/cmcuza/TerseTS/blob/doc/improve-readme/bindings/python/tersets/__init__.py#L37).

Reply:
Fixed.